bigquery/main.go

```diff
@@ -83,52 +101,75 @@ func (bq *bqNotifier) SetUp(ctx context.Context, cfg *notifiers.Config, _ notifi
 	}
 
 	// Initialize client
-	bq.filter = prd
-	bq.client, err = bigquery.NewClient(ctx, projectID)
+	n.filter = prd
+	n.client, err = n.bqf.Make(ctx)
+	if err != nil {
+		return err
+	}
+
 	if err != nil {
 		return fmt.Errorf("Failed to initialize bigquery client: %v", err)
 	}
 
 	// Extract dataset id and table id from config
-	rs := rgp.FindStringSubmatch(parsed)
-	bq.dataset = bq.client.Dataset(rs[1])
-	bq.table = bq.dataset.Table(rs[2])
+	rs := tableResource.FindStringSubmatch(parsed)
+	if len(rs) != 3 {
+		return fmt.Errorf("Failed to parse valid table URI: %v", parsed)
+	}
+	err = n.client.EnsureDataset(ctx, rs[1])
+	if err != nil {
+		return err
+	}
```

Comment:
Fix: prefer the `if err := foo(); err != nil` style whenever possible.

Reply:
Resolved!